linux_docker_resources/Dockerfile-RHEL

```diff
@@ -59,24 +60,37 @@ RUN dnf update \
 RUN rosdep init && rosdep update
 
 # Install a newer version of pytest
-RUN python3 -m pip install -U pytest pytest-rerunfailures
+RUN if test ${EL_RELEASE} = 8; then python3 -m pip install -U pytest pytest-rerunfailures; fi
+
+# Get some missing packages from pip
+RUN if test ${EL_RELEASE} = 9; then \
+    dnf install \
+      gcc \
+      graphviz-devel \
+      -y && \
+    python3 -m pip install \
+      matplotlib \
+      pygraphviz; \
+  fi
 
 # These dependencies were generated by invoking rosdep on the latest version of the target workspace:
 # rosdep install --from-paths src --ignore-src --simulate --reinstall --skip-keys ...
 RUN dnf install \
     CUnit-devel \
     acl \
-    assimp-devel \
+    $(if test ${EL_RELEASE} != 8; then echo asio-devel; fi) \
+    $(if test ${EL_RELEASE} != 9; then echo assimp-devel; fi) \
```

Comment:
I know you were trying to make this opportunistic, but it feels like this scatters around a lot of if statements.  Might it make more sense to three blocks of `dnf install` commands:

1. Package names that are common to both EL8 and EL9
2. Package names that are only for EL8
3. Package names that are only for EL9

?

Reply:
Eh, I'm lukewarm about this. These inline conditionals aren't particularly long or verbose (especially as far as bash is concerned), and it helps to avoid creating new and short-lived blocks.

I find this approach more maintainable, and I'll probably be doing most of the maintenance. Not a hill I'm willing to die on, but I'd prefer it.

Reply:
I don't feel that strongly about it, so if you prefer this, that is OK with me.